code/game/machinery/stasis.dm

```diff
@@ -0,0 +1,138 @@
+#define STASIS_TOGGLE_COOLDOWN 50
+/obj/machinery/stasis
+	name = "Lifeform Stasis Unit"
+	desc = "A not so comfortable looking bed with some nozzles at the top and bottom. It will keep someone in stasis."
+	icon = 'icons/obj/machines/stasis.dmi'
+	icon_state = "stasis"
+	density = FALSE
+	can_buckle = TRUE
+	buckle_lying = 90
+	circuit = /obj/item/circuitboard/machine/stasis
+	idle_power_usage = 40
+	active_power_usage = 340
+	fair_market_price = 10
+	payment_department = ACCOUNT_MED
+	var/stasis_enabled = TRUE
+	var/last_stasis_sound = FALSE
+	var/stasis_can_toggle = 0
+	var/mattress_state = "stasis_on"
+	var/obj/effect/overlay/vis/mattress_on
+
+/obj/machinery/stasis/Initialize(mapload)
+	if(mattress_state)
+		mattress_on = SSvis_overlays.add_vis_overlay(src, icon, mattress_state, layer, plane, dir, alpha = 0, unique = TRUE)
+	return ..()
+
+/obj/machinery/stasis/examine(mob/user)
+	..()
+	var/turn_on_or_off = stasis_enabled ? "turn off" : "turn on"
+	to_chat(user, "<span class='notice'>Alt-click to [turn_on_or_off] the machine.</span>")
+
+/obj/machinery/stasis/proc/play_power_sound()
+	var/_running = stasis_running()
+	if(last_stasis_sound != _running)
+		var/sound_freq = rand(5120, 8800)
+		if(_running)
+			playsound(src, 'sound/machines/synth_yes.ogg', 50, TRUE, frequency = sound_freq)
+		else
+			playsound(src, 'sound/machines/synth_no.ogg', 50, TRUE, frequency = sound_freq)
+		last_stasis_sound = _running
+
+/obj/machinery/stasis/AltClick(mob/user)
+	if(world.time >= stasis_can_toggle && user.canUseTopic(src))
+		stasis_enabled = !stasis_enabled
+		stasis_can_toggle = world.time + STASIS_TOGGLE_COOLDOWN
+		playsound(src, 'sound/machines/click.ogg', 60, TRUE)
+		play_power_sound()
+		update_icon()
+
+/obj/machinery/stasis/Exited(atom/movable/AM, atom/newloc)
+	if(AM == occupant)
+		var/mob/living/L = AM
+		if(L.IsInStasis())
+			thaw_them(L)
+	. = ..()
+
+/obj/machinery/stasis/proc/stasis_running()
+	return stasis_enabled && is_operational()
+
+/obj/machinery/stasis/update_icon()
+	. = ..()
+	var/_running = stasis_running()
+
+	if(mattress_on.alpha ? !_running : _running) //check the inverse of _running compared to truthy alpha, to see if they differ
+		var/new_alpha = _running ? 255 : 0
+		var/easing_direction = _running ? EASE_OUT : EASE_IN
+		animate(mattress_on, alpha = new_alpha, time = 50, easing = CUBIC_EASING|easing_direction)
+
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays - mattress_on)
```

Comment:
this is not valid code

doing managed_vis_overlays - mattress_on is causing a type mismatch

Reply:
this runtime can be triggered by loading the emergency medical deck holodeck

Reply:
I deserve it for expecting managed_vis_overlays to always be there. The holodeck removes it https://github.com/tgstation/tgstation/blob/18bc177511551e0d0569332969429211ece02429/code/modules/holodeck/area_copy.dm#L17-L25

I didn't do safety checks because I wanted runtimes if it didn't work, since I expected it to exist, as a sort of assertion, but now I know that there is something that can remove it during normal operation.